packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewHelper.kt

```diff
@@ -108,6 +108,29 @@ object CameraViewHelper {
     return exifMap
   }
 
+  @JvmStatic
+  fun setExifData(baseExif: ExifInterface, exifMap: Map<String, Any>) {
+    for ((_, name) in exifTags) {
+      exifMap[name]?.let {
+        // Convert possible type to string before putting into baseExif
+        when(it::class.simpleName) {
+          "String" -> baseExif.setAttribute(name, it as String)
+          "Double" -> baseExif.setAttribute(name, (it as Double).toBigDecimal().toPlainString())
+          "Int" -> baseExif.setAttribute(name, (it as Int).toBigDecimal().toPlainString())
+          "Boolean" -> baseExif.setAttribute(name, (it as Boolean).toString())
+        }
+      }
+    }
+
+    if (exifMap.containsKey(ExifInterface.TAG_GPS_LATITUDE) && exifMap.containsKey(ExifInterface.TAG_GPS_LONGITUDE)) {
+      baseExif.setLatLong(exifMap[ExifInterface.TAG_GPS_LATITUDE] as Double,
+        exifMap[ExifInterface.TAG_GPS_LONGITUDE] as Double)
+    }
+    if(exifMap.containsKey(ExifInterface.TAG_GPS_ALTITUDE)){
+      baseExif.setAltitude(exifMap[ExifInterface.TAG_GPS_ALTITUDE] as Double)
+    }
```

Comment:
What happens if `exifMap[ExifInterface.TAG_GPS_LONGITUDE]` is not `Double` e.g. somebody provides a string here?

Reply:
I have added type check and also add a exception handling in case the value is invalid coordinates.